### PR TITLE
Add breadcrumb support

### LINKED ISSUE #2: Add Breadcrumbs support
Like many other plugins it would be nice to have option to show breadcrumbs that are describing current level in a hierarchy of objects.

example how it works for xml file
image

Ideal would be if breadcrumbs would be highlighted in diferent colors
image

### PR DESCRIPTION
**Pull request by SamoSchneider**
resolves #2 

Screenshot from 2021-11-04 12-48-51

Signed-off-by: Samuel Schneider <samuel.schneider@pantheon.tech>

### CODE REVIEW

File: intelij-plugin/src/main/java/tech/pantheon/yanginator/plugin/breadcrumbs/YangBreadcrumbsPresentationProvider.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package tech.pantheon.yanginator.plugin.breadcrumbs;
+
+import com.intellij.psi.PsiElement;
+import com.intellij.ui.JBColor;
+import com.intellij.xml.breadcrumbs.BreadcrumbsPresentationProvider;
+import com.intellij.xml.breadcrumbs.CrumbPresentation;
+
+public class YangBreadcrumbsPresentationProvider extends BreadcrumbsPresentationProvider {
+    private final CrumbPresentation[] colors = {new YangCrumbPresentation(JBColor.BLUE),
```
Comment (PetkoNosal) [inline intelij-plugin/src/main/java/tech/pantheon/yanginator/plugin/breadcrumbs/YangBreadcrumbsPresentationProvider.java:9]: this should be static and also much better if like this:
```
        private static final CrumbPresentation[] COLORS = {
            new YangCrumbPresentation(JBColor.BLUE),
            new YangCrumbPresentation(JBColor.GREEN),
            new YangCrumbPresentation(JBColor.RED)
    };
```

File: intelij-plugin/src/main/java/tech/pantheon/yanginator/plugin/breadcrumbs/YangBreadcrumbsProvider.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    @Override
+    public boolean acceptElement(@NotNull PsiElement psiElement) {
+        return psiElement instanceof YangListStmt ||
+                psiElement instanceof YangContainerStmt;
+    }
+
+    @Override
+    public @NotNull
```
Comment (PetkoNosal) [inline intelij-plugin/src/main/java/tech/pantheon/yanginator/plugin/breadcrumbs/YangBreadcrumbsProvider.java:29]: visibility modifier (that public) should be on same line as method declaration

File: intelij-plugin/src/main/java/tech/pantheon/yanginator/plugin/breadcrumbs/YangCrumbPresentation.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package tech.pantheon.yanginator.plugin.breadcrumbs;
+
+import com.intellij.ui.JBColor;
+import com.intellij.xml.breadcrumbs.CrumbPresentation;
+
+import java.awt.*;
```
Comment (PetkoNosal) [inline intelij-plugin/src/main/java/tech/pantheon/yanginator/plugin/breadcrumbs/YangCrumbPresentation.java:6]: please replace * imports with explicitly import only what should be imported

Comment (ifoltin) [inline intelij-plugin/src/main/java/tech/pantheon/yanginator/plugin/breadcrumbs/YangCrumbPresentation.java:6]: Do not use wildcard imports, please.

**Review by PetkoNosal — CHANGES_REQUESTED** [review]
Small teaks and changes needed!

**Review by PetkoNosal — APPROVED** [review]

**Pull request merged**